src/mame/namco/namcos23.cpp

```diff
@@ -2069,12 +2065,10 @@ void namcos23_state::c435_pio_w(uint16_t data)
 	}
 
 	if(!known) {
-		char buf[4096];
-		char *p = buf;
-		p += sprintf(p, "c435 -");
+		std::string buf = "c435 -";
 		for(int i=0; i<m_c435_buffer_pos; i++)
-			p += sprintf(p, " %04x", m_c435_buffer[i]);
-		p += sprintf(p, "\n");
+			buf += util::string_format(" %04x", m_c435_buffer[i]);
+		buf += "\n";
 		logerror(buf);
 	}
```

Comment:
Did you see the benchmarking I did previously: https://github.com/mamedev/mame/pull/10933#discussion_r1118070558

It's not directly the same in this case, but I suspect stream may be slower. My point previously is, we should just pick what is most readable source code, and not second guess performance.

Having said that if you think stream is more readable then happy to change.

Reply:
But `util::sting_format` will internally create a stream object anyway.  Is constructing and destructing `m_c435_buffer_pos` temporary stream objects (one for each loop iteration) really going to be faster than using a single stream object and using `util::stream_format` on it multiple times?

I’ll admit I haven’t extensively tested how GCC and clang deal with string concatenations for a few years.  I know Java compilers have recognised and special-cased string concatenations to avoid the intermediate temporary objects for over a decade now – it’s possible the C++ compilers are doing something similar now.

Reply:
Gone with stream.